lib/Conversion/ExportVerilog/PrepareForEmission.cpp

```diff
@@ -467,9 +492,15 @@ static bool hoistNonSideEffectExpr(Operation *op) {
 
 /// Check whether an op is a declaration that can be moved.
 static bool isMovableDeclaration(Operation *op) {
-  return op->getNumResults() == 1 &&
-         op->getResult(0).getType().isa<InOutType, sv::InterfaceType>() &&
-         op->getNumOperands() == 0;
+  if (op->getNumResults() != 1 ||
+      !op->getResult(0).getType().isa<InOutType, sv::InterfaceType>())
+    return false;
+
+  // If all operands (e.g. init value) are constant, it is safe to move
```

Comment:
This is very conservative, but probably fine for now.

Constants in general should be hoisted to the front of the block (and are at various places).